### PR TITLE
docs: add constructor docs for each provider and account type

### PR DESCRIPTION
**Pull request by avasisht23**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, see guidleines below)
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:fix`)
- [x] Is the base branch you're merging into `development` and not `main`?

---

## PR-Codex overview
This PR focuses on removing default values for certain properties in the code. 

### Detailed summary
- Removed default value for `baseFeeBufferPercent` property in `schema.ts`.
- Removed default value for `maxPriorityFeeBufferPercent` property in `schema.ts`.
- Removed default value for `txMaxRetries` property in `schema.ts`.
- Removed default value for `txRetryIntervalMs` property in `schema.ts`.
- Removed default value for `txRetryMulitplier` property in `schema.ts`.
- Removed default value for `minPriorityFeePerBid` property in `schema.ts`.
- Added constructor for `SmartAccountProvider` in `config.ts`.
- Added constructor for `EthersProviderAdapter` in `config.ts`.
- Added constructor for `EthersProviderAdapter` in `provider-adapter/constructor.md`.
- Added constructor for `SmartAccountProvider` in `provider/constructor.md`.
- Added constructor for `SimpleSmartContractAccount` in `accounts/constructor.md`.
- Added constructor for `LightSmartContractAccount` in `accounts/constructor.md`.

> The following files were skipped due to too many changes: `site/packages/aa-accounts/light-account/constructor.md`, `site/packages/aa-alchemy/provider/constructor.md`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

image
image
image
image
image

### CODE REVIEW

**Comment by avasisht23** [comment]
Current dependencies on/for this PR:
* development
  * **PR #203** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/203" target="_blank">Graphite</a> 
    * **PR #202** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/202" target="_blank">Graphite</a> 
      * **PR #201** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/201" target="_blank">Graphite</a> 
        * **PR #204** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/204" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/204?utm_source=stack-comment).

File: site/packages/aa-accounts/light-account/constructor.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+## Returns
+
+### `LightSmartContractAccount`
+
+A new instance of an `LightSmartContractAccount`.
+
+## Parameters
+
+### `params: SimpleSmartAccountParams`
```
Comment (avasisht23) [inline site/packages/aa-accounts/light-account/constructor.md:63]: intentional, per comment above

File: packages/core/src/provider/schema.ts [hunk 2]
```diff
@@ -9,22 +9,22 @@ export const SmartAccountProviderOptsSchema = z.object({
...
   /**
    * The mulitplier on interval length to wait between retries while waiting for tx receipts (default: 1.5)
    */
-  txRetryMulitplier: z.number().optional().default(1.5),
+  txRetryMulitplier: z.number().optional(),
 
   /**
    * used when computing the fees for a user operation (default: 100_000_000n)
    */
-  minPriorityFeePerBid: z.bigint().optional().default(100_000_000n),
```
Comment (denniswon) [inline packages/core/src/provider/schema.ts:27]: why defaults removed? on the comment, it still says defaults

Comment (avasisht23) [reply packages/core/src/provider/schema.ts:27]: image

Yeah sadly the default made these fields required even though they're optional.

Defaults are still set in the constructor.

Comment (denniswon) [reply packages/core/src/provider/schema.ts:27]: Yeah had the same with entrypoint

**Review by denniswon — APPROVED** [review]

**Pull request merged**